src/main/java/com/spartronics4915/frc2019/subsystems/RobotStateEstimator.java

```diff
@@ -70,27 +103,31 @@ public void registerEnabledLoops(ILooper looper)
         @Override
         public synchronized void onStart(double timestamp)
         {
-            left_encoder_prev_distance_ = drive_.getLeftEncoderDistance();
-            right_encoder_prev_distance_ = drive_.getRightEncoderDistance();
+            mLeftEncoderPrevDistance = mDrive.getLeftEncoderDistance();
+            mRightEncoderPrevDistance = mDrive.getRightEncoderDistance();
 
         }
 
         @Override
         public synchronized void onLoop(double timestamp)
         {
-            final double left_distance = drive_.getLeftEncoderDistance();
-            final double right_distance = drive_.getRightEncoderDistance();
-            final double delta_left = left_distance - left_encoder_prev_distance_;
-            final double delta_right = right_distance - right_encoder_prev_distance_;
-            final Rotation2d gyro_angle = drive_.getHeading();
-            final Twist2d odometry_velocity = robot_state_.generateOdometryFromSensors(
-                    delta_left, delta_right, gyro_angle);
-            final Twist2d predicted_velocity = Kinematics.forwardKinematics(drive_.getLeftLinearVelocity(),
-                    drive_.getRightLinearVelocity());
-            robot_state_.addObservations(timestamp, odometry_velocity,
-                    predicted_velocity);
-            left_encoder_prev_distance_ = left_distance;
-            right_encoder_prev_distance_ = right_distance;
+            final double leftDistance = mDrive.getLeftEncoderDistance();
+            final double rightDistance = mDrive.getRightEncoderDistance();
+            final double leftDelta = leftDistance - mLeftEncoderPrevDistance;
+            final double rightDelta = rightDistance - mRightEncoderPrevDistance;
+            final Rotation2d gyroAngle = mDrive.getHeading();
+
+            mLeftEncoderPrevDistance = leftDistance;
+            mRightEncoderPrevDistance = rightDistance;
+
+            final Twist2d measured_velocity = getVelocityFromDeltas(
+                    leftDelta, rightDelta, gyroAngle);
+            final Twist2d predicted_velocity = Kinematics.forwardKinematics(mDrive.getLeftLinearVelocity(),
+                    mDrive.getRightLinearVelocity());
+
+            mEncoderRobotState.addObservations(timestamp,
+                Kinematics.integrateForwardKinematics(mEncoderRobotState.getLatestFieldToVehicle().getValue(), measured_velocity),
```

Comment:
hm - orig didn't invoke integrateFwdKinematics... Not sure if its wrong or right (seems like it might be right), but I'm just curious about this diff.

Reply:
The original actually does invoke it [here](https://github.com/Spartronics4915/Spartronics-254Base2018/blob/20cec7227fdd4634471beab0e489b24dd61c93a4/src/main/java/com/spartronics4915/frc2019/RobotState.java#L102).